identity-hub-did/src/main/java/org/eclipse/dataspaceconnector/identityhub/did/IdentityHubCredentialsVerifier.java

```diff
@@ -0,0 +1,104 @@
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.identityhub.did;
+
+import com.nimbusds.jwt.SignedJWT;
+import org.eclipse.dataspaceconnector.iam.did.spi.credentials.CredentialsVerifier;
+import org.eclipse.dataspaceconnector.iam.did.spi.document.DidDocument;
+import org.eclipse.dataspaceconnector.identityhub.client.IdentityHubClient;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.result.AbstractResult;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Obtains and verifies credentials associated with a DID.
+ * The DID document contains an IdentityHub service, the IdentityHubCredentialsVerifier gets credentials from the
+ * IdentityHub instance and verifies the credentials.
+ */
+public class IdentityHubCredentialsVerifier implements CredentialsVerifier {
+
+    private final IdentityHubClient identityHubClient;
+    private final Monitor monitor;
+    private final SignatureVerifier signatureVerifier;
+
+    /**
+     * Create a new credential verifier that uses an Identity Hub
+     *
+     * @param identityHubClient IdentityHubClient.
+     */
+    public IdentityHubCredentialsVerifier(IdentityHubClient identityHubClient, Monitor monitor, SignatureVerifier signatureVerifier) {
+        this.identityHubClient = identityHubClient;
+        this.monitor = monitor;
+        this.signatureVerifier = signatureVerifier;
+    }
+
+    /**
+     * Get credentials from the IdentityHub of participant, verifies the credentials, and returns the successfully verified credentials.
+     * To be successfully verified, credentials needs to be signed by the specified issuer in the JWT.
+     *
+     * @param didDocument of a participant. The Did Document should contain an IdentityHub service.
+     * @return VerifiableCredentials.
+     */
+    @Override
+    public Result<Map<String, Object>> getVerifiedCredentials(DidDocument didDocument) {
+        var hubBaseUrl = getIdentityHubBaseUrl(didDocument);
+        if (hubBaseUrl.failed()) return Result.failure(hubBaseUrl.getFailureMessages());
+
+        var jwts = identityHubClient.getVerifiableCredentials(hubBaseUrl.getContent());
+        if (jwts.failed()) {
+            return Result.failure(jwts.getFailureMessages());
+        }
+        var verifiedClaims = jwts.getContent()
+                .stream()
+                .filter(signatureVerifier::isSignedByIssuer);
+        var claims = verifiedClaims.map(this::extractCredential)
+                .filter(AbstractResult::succeeded)
+                .map(AbstractResult::getContent)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        return Result.success(claims);
+    }
+
+    private Result<Map.Entry<String, Object>> extractCredential(SignedJWT jwt) {
```

Comment:
Extracting the claims from a `SignedJWT` is a common reusable functionality that could be placed in a utility class ([same functionality](https://github.com/agera-edc/IdentityHub/blob/e456e4bfad2dc15f97c22a16eb4485cb43ff179f/client-cli/src/main/java/org/eclipse/dataspaceconnector/identityhub/cli/ListVerifiableCredentialsCommand.java#L55) is used in the CLI for instance). Consider moving this logic to a central place.